Write console output to dsclog file.

BerheAbrha opened the pull request:
- [x] Include the console output of InstallModule.py and RemoveModule.py in the dsclog file.
-[x] Update misleading error message. 

Example of trying to install **unsigned module.**
```
Extracting to /opt/microsoft/omsconfig/modules
Error: Cannot find asc file as part of module

```

InstallModule.py Log in omsconfig.log.
```
Extracting to /opt/microsoft/omsconfig/modules
Installing resource MSFT_nxFileInventoryResource
Extracting to /opt/microsoft/omsconfig/modules
Installing resource MSFT_nxOMSPluginResource
Extracting to /opt/microsoft/omsconfig/modules
Installing resource MSFT_nxOMSSyslogResource
Extracting to /opt/microsoft/omsconfig/modules
Installing resource MSFT_nxOMSPerfCounterResource
Extracting to /opt/microsoft/omsconfig/modules
Installing resource MSFT_nxOMSGenerateInventoryMofResource
Extracting to /opt/microsoft/omsconfig/modules
Installing resource MSFT_nxOMSAutomationWorkerResource
Extracting to /opt/microsoft/omsconfig/modules
Installing resource MSFT_nxOMSAuditdPluginResource
```
Resources log.
```
2018/03/20 00:46:14: INFO: Scripts/nxOMSPlugin.pyc(89):
OMSAgent is multi-homed and resource is updating workspace 2843c125-5a7b-4820-bf83-9ed4b6b7046c
2018/03/20 00:46:14: INFO: Scripts/nxOMSPlugin.pyc(89):
OMSAgent is multi-homed and resource is updating workspace 2843c125-5a7b-4820-bf83-9ed4b6b7046c
2018/03/20 00:46:15: INFO: Scripts/nxOMSSyslog.pyc(56):
Config file is /etc/opt/omi/conf/omsconfig/rsyslog-oms.conf.
2018/03/20 00:46:15: INFO: Scripts/nxOMSSyslog.pyc(189):
Successfully read /etc/rsyslog.d/95-omsagent.conf.
2018/03/20 00:46:15: INFO: Scripts/nxOMSSyslog.pyc(458):
Succesfully read /etc/opt/microsoft/omsagent/2843c125-5a7b-4820-bf83-9ed4b6b7046c/conf/omsagent.d/syslog.conf for syslog port.
2018/03/20 00:46:15: INFO: Scripts/nxOMSSyslog.pyc(56):
Config file is /etc/opt/omi/conf/omsconfig/rsyslog-oms.conf.
2018/03/20 00:46:15: INFO: Scripts/nxOMSSyslog.pyc(189):
Successfully read /etc/rsyslog.d/95-omsagent.conf.
2018/03/20 00:46:15: INFO: Scripts/nxOMSSyslog.pyc(458):
Succesfully read /etc/opt/microsoft/omsagent/2843c125-5a7b-4820-bf83-9ed4b6b7046c/conf/omsagent.d/syslog.conf for syslog port.
2018/03/20 00:46:15: INFO: Scripts/nxOMSSyslog.pyc(223):
Successfully read /etc/rsyslog.d/95-omsagent.conf.
2018/03/20 00:46:15: INFO: Scripts/nxOMSSyslog.pyc(458):
```

Also,

- [x] I have verified the python script work with versions of python  2.4, 2.7 and  3.0.

narrieta commented:
>  Include the console output of InstallModule.py and RemoveModule.py in the dsclog file.

How about other scripts? (e.g. Register.py)

BerheAbrha commented:
InstallModule.py and RemoveModule.py  are the only python scripts that are being called directly from c/c++. Register.py is not being called from DSC code. (probably the extension is calling it).

---
In reply to: [374392116](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/368#issuecomment-374392116) [](ancestors = 374392116)

narrieta commented on LCM/scripts/InstallModule.py:
```diff
@@ -11,6 +11,12 @@
     import zipfile
 
+scriptFolderPath = os.path.dirname(os.path.realpath(__file__))
+fullPathDSCLogger = os.path.join(scriptFolderPath, 'nxDSCLog.py')
```
we are introducing a dependency of the engine (pull client) on a resource. that doesn't sound correct... maybe some refactoring so that the resource and pull client depend on some component in the engine?

BerheAbrha replied:
nxDSCLog.py is not actually a resource. It is just a helper script used by all the resources for logging.

narrieta commented:
is the output of Register.py being capture in any log?

BerheAbrha commented:
I just checked.  It should be part of the extension logs. see here from the extension repo.
[https://github.com/Azure/azure-linux-extensions/blob/67cf2686f9536c5c97c013cedc600f02c50f3675/DSC/dsc.py#L338](https://github.com/Azure/azure-linux-extensions/blob/67cf2686f9536c5c97c013cedc600f02c50f3675/DSC/dsc.py#L338)


---
In reply to: [374399390](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/368#issuecomment-374399390) [](ancestors = 374399390)

BerheAbrha commented:
@narrieta, Let me know  if you have unaddressed comment.  @kwirkykat  I have verified the same script work in python 2.7 and python 3.0. ( have not verified with 2.4 but I not using any language feature that does not exist on the script already). please review if you have further comment.

kwirkykat commented on LCM/scripts/InstallModule.py:
```diff
@@ -5,14 +5,17 @@
 import shutil
 import subprocess
 import sys
+import platform
```
This is also a duplicated import
